src/lib/AST/ASTVisitor.cpp

```diff
@@ -760,40 +889,45 @@ class ASTVisitor
         which were implicitly instantiated, this will be whichever `Decl`
         was used as the pattern for instantiation.
     */
-    template<typename DeclTy>
-    DeclTy* getInstantiatedFrom(DeclTy* D);
+    template <class DeclTy>
+    DeclTy*
+    getInstantiatedFrom(DeclTy* D);
 
-    template<typename DeclTy>
+    template <class DeclTy>
```

Comment:
Please leave template-heads in their current style (no space between template and '<', and using `typename`)

Reply:
I originally used `class` but someone started using `typename`. Not clear why the longer version is preferred. Also Alan are you using clang-format? Because it sure looks like it.

Reply:
> I originally used class but someone started using typename. 

Yes. You told me to always use `class` because it was shorter.

> Not clear why the longer version is preferred. 

I don't think it was an explicit decision. I think it's just something Krystian likes. So I reverted it.

> Also Alan are you using clang-format? Because it sure looks like it.

Not at all. If I used clang-format at this point this commit would have 4000 changes.